ICRCs/ICRC-97/ICRC-97.md

```diff
@@ -0,0 +1,101 @@
+
+ICRC-7 does not specify the representation of token metadata any further than that it is represented in a generic manner
+as a vector of (text, Value)-pairs. The ICRC-97 standard defines a token metadata
+standard for ICRC-7 and other NFT standards.
+
+### Entrypoint
+
+The following metadata property MUST be defined in the root of the token metadata if we want to return on-chain
+metadata.
+
+| Root Metadata Property | ICRC-3 Type     | Description                            |
+|------------------------|-----------------|----------------------------------------|
+| icrc97:metadata        | variant { Map } | Contains on-chain metadata properties. |
+
+Alternatively the following metadata property MUST be defined in the root of the token metadata if we want to return
+external
+off-chain metadata instead of on-chain metadata.
+
+| Root Metadata Property   | ICRC-3 Type     | Description                                       |
+|--------------------------|-----------------|---------------------------------------------------|
+| icrc97:external_metadata | variant { Map } | Contains below external JSON metadata properties. |
+
+| External Metadata Property | Optional | ICRC-3 Type      | Description                                                                    |
+|----------------------------|----------|------------------|--------------------------------------------------------------------------------|
+| url                        | No       | variant { Text } | URL that returns the metadata in JSON format (protocol is not limited to HTTP) |
+| sha256_hash                | Yes      | variant { Blob } | SHA-256 hash of HTTP response body bytes returned from above url.              |
+
+### Metadata properties
+
+Below table lists all (optional) metadata properties defined in this standard. Each property is defined in both ICRC-3
+and JSON type to support both on-chain `Value` and off-chain JSON metadata, respectively.
+
+| Metadata Property | Optional | ICRC-3 Type                             | JSON Type | Description                                             |
+|-------------------|----------|-----------------------------------------|-----------|---------------------------------------------------------|
+| external_url      | Yes      | variant { Text }                        | string    | URL that allows the user to view the item on your site. |
+| name              | Yes      | variant { Text }                        | string    | Plain text.                                             |
+| description       | Yes      | variant { Text }                        | string    | Markdown.                                               |
+| assets            | Yes      | variant { Array = vec variant { Map } } | object[]  | List of assets ordered by priority descending.          |
+| attributes        | Yes      | variant { Array = vec variant { Map } } | object[]  | List of attributes ordered by priority descending.      |
```

Comment:
are the attributes expected to be immutable? otherwise we might want to distinguish between mutable and immutable attributes

Reply:
the term "immutable" is of course tricky, given the nature of canisters living on ICP.

did you generally discuss to track the history of metadata changes so far? 🤔

Reply:
Good points, there's the [update schema in ICRC-7](https://github.com/dfinity/ICRC/blob/icrc7-wg-draft/ICRCs/ICRC-7/ICRC-7.md#icrc7_update-block-schema) for the purpose of tracking metadata updates. I'll mention how to make the metadata immutable, basically by not returning above block type in [ICRC-3 supported block types](icrc3_supported_block_types). I'll also mention on how ICRC-97 metadata updates should be encoded as blobs in the update schema in case the metadata is not immutable.

Reply:
I think I need to read more about ICRC-3 🤔 

note: the link is not working.

Reply:
> note: the link is not working.

Fixed